# Cursos de medicina no Brasil

%md
No Brasil, os cursos de medicina são extremamente concorridos e exigentes, com uma carga horária intensa e duração média de seis anos. As faculdades de medicina no país oferecem formação em diversas áreas, como cirurgia, pediatria, ginecologia e obstetrícia, entre outras especialidades médicas.

Existem opções de bolsas para estudantes que desejam cursar medicina no país. Uma das principais opções é o Programa Universidade para Todos (PROUNI), que oferece bolsas de estudos parciais e integrais em universidades particulares para estudantes de baixa renda que tenham obtido boas notas no Enem.

Além disso, há também a possibilidade de ingressar em universidades públicas, que são gratuitas e oferecem um ensino de qualidade reconhecido mundialmente. No entanto, o processo seletivo para ingresso em universidades públicas é bastante concorrido e exige um alto desempenho dos candidatos.

Nas próximas seções, iremos disponibilizar alguns resultados com base em dados modificados do PROUNI e de universidades públicas, a fim de fornecer informações mais detalhadas sobre as opções de bolsas e ingresso nessas instituições de ensino.

Esse conjunto de dados apresenta informações sobre cursos de graduação na área de saúde em universidades brasileiras. As colunas incluem informações sobre o grau do curso, turno, mensalidade, tipos de bolsa integral e parcial (cotas e ampla concorrência), cidade e universidade. Além disso, há informações sobre as notas de corte para acesso às vagas com bolsa e sem bolsa.

Algumas informações que podem ser interessantes avaliar nesse conjunto de dados são: quais universidades oferecem bolsas de estudos e qual é a porcentagem de bolsas concedidas para cada curso; quais são as notas de corte para acesso às vagas com bolsa e sem bolsa; quais são as universidades e cidades com os cursos mais caros e mais baratos em relação à mensalidade; quais são as opções de cursos em diferentes cidades e estados do país.

In [0]:
# Instalar requests (se necessário)
%pip install requests


Python interpreter will be restarted.
Python interpreter will be restarted.


In [0]:
import requests

# URL do arquivo
url = "https://raw.githubusercontent.com/alura-cursos/pandasspark/refs/heads/main/dados/cursos-prouni.csv"

# Fazer o download do arquivo
response = requests.get(url)

# Escrever o arquivo no DBFS usando dbutils
dbutils.fs.put("/FileStore/cursos-prouni.csv", response.content.decode('utf-8'), overwrite=True)


Wrote 9225477 bytes.
Out[1]: True

In [0]:
local_do_arquivo = '/FileStore/cursos-prouni.csv'

In [0]:
"""
# Ler o arquivo CSV que foi carregado no DBFS
df = spark.read.csv("/FileStore/cursos-prouni.csv", header=True)

# Mostrar os dados
df.show()
"""

Out[3]: '\n# Ler o arquivo CSV que foi carregado no DBFS\ndf = spark.read.csv("/FileStore/cursos-prouni.csv", header=True)\n\n# Mostrar os dados\ndf.show()\n'

In [0]:
import requests

# URL do arquivo
url = "https://raw.githubusercontent.com/alura-cursos/pandasspark/refs/heads/main/dados/reajuste.csv"

# Fazer o download do arquivo
response = requests.get(url)

# Escrever o arquivo no DBFS usando dbutils
dbutils.fs.put("/FileStore/reajuste.csv", response.content.decode('utf-8'), overwrite=True)


Wrote 124990 bytes.
Out[4]: True

In [0]:
'''
# Ler o arquivo CSV que foi carregado no DBFS
df_reajuste = spark.read.csv("/FileStore/reajuste.csv", header=True)

# Mostrar os dados
df_reajuste.show()
'''

Out[5]: '\n# Ler o arquivo CSV que foi carregado no DBFS\ndf_reajuste = spark.read.csv("/FileStore/reajuste.csv", header=True)\n\n# Mostrar os dados\ndf_reajuste.show()\n'

In [0]:
import pyspark.pandas as ps

df = ps.read_csv(local_do_arquivo, index_col = 'curso_id')

Esse conjunto de dados apresenta informações sobre cursos de graduação na área de saúde em universidades brasileiras. As colunas incluem informações sobre o grau do curso, turno, mensalidade, tipos de bolsa integral e parcial (cotas e ampla concorrência), cidade e universidade. Além disso, há informações sobre as notas de corte para acesso às vagas com bolsa e sem bolsa.

Algumas informações que podem ser interessantes avaliar nesse conjunto de dados são: quais universidades oferecem bolsas de estudos e qual é a porcentagem de bolsas concedidas para cada curso; quais são as notas de corte para acesso às vagas com bolsa e sem bolsa; quais são as universidades e cidades com os cursos mais caros e mais baratos em relação à mensalidade; quais são as opções de cursos em diferentes cidades e estados do país.

In [0]:
local_do_arquivo2 = '/FileStore/reajuste.csv'
df_mensalidade = ps.read_csv(local_do_arquivo2)

In [0]:
df.head()

grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 curso_busca 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 curso_id 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 706710394154 
 Bacharelado 
 Integral 
 9999.99 
 15 
 14.0 
 NaN 
 NaN 
 Medicina 
 Campo Grande 
 MS 
 NTAwMjAwNDAyNzA0 
 Universidade Anhanguera - UNIDERP 
 CAMPO GRANDE - SEDE - Miguel Couto 
 706710 
 Medicina 
 740.22 
 726.46 
 NaN 
 NaN 
 
 
 104191210567043 
 Bacharelado 
 Noturno 
 9836.40 
 1 
 NaN 
 NaN 
 NaN 
 Enfermagem 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Enfermagem 
 663.36 
 NaN 
 NaN 
 NaN 
 
 
 1002328574024 
 Bacharelado 
 Integral 
 9715.61 
 2 
 5.0 
 6.0 
 10.0 
 Medicina 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 104191212798093 
 Bacharelado 
 Noturno 
 9689.34 
 3 
 2.0 
 NaN 
 NaN 
 Psicologia 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Psicologia 
 651.00 
 652.22 
 NaN 
 NaN 
 
 
 65899611932754 
 Bacharelado 
 Integral 
 9674.34 
 4 
 1.0 
 5.0 
 2.0 
 Medicina 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26

In [0]:
df.shape

Out[9]: (41447, 19)

In [0]:
df['curso_busca'].equals(df['nome']).sum()

Out[10]: 41447

In [0]:
df = df.drop(['curso_busca'], axis = 1)

In [0]:
df.head()

grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 curso_id 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 706710394154 
 Bacharelado 
 Integral 
 9999.99 
 15 
 14.0 
 NaN 
 NaN 
 Campo Grande 
 MS 
 NTAwMjAwNDAyNzA0 
 Universidade Anhanguera - UNIDERP 
 CAMPO GRANDE - SEDE - Miguel Couto 
 706710 
 Medicina 
 740.22 
 726.46 
 NaN 
 NaN 
 
 
 104191210567043 
 Bacharelado 
 Noturno 
 9836.40 
 1 
 NaN 
 NaN 
 NaN 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Enfermagem 
 663.36 
 NaN 
 NaN 
 NaN 
 
 
 1002328574024 
 Bacharelado 
 Integral 
 9715.61 
 2 
 5.0 
 6.0 
 10.0 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 104191212798093 
 Bacharelado 
 Noturno 
 9689.34 
 3 
 2.0 
 NaN 
 NaN 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Psicologia 
 651.00 
 652.22 
 NaN 
 NaN 
 
 
 65899611932754 
 Bacharelado 
 Integral 
 9674.34 
 4 
 1.0 
 5.0 
 2.0 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26

In [0]:
df = df.rename(columns  = {'nome':'nome_curso'})

In [0]:
df.head()

grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 curso_id 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 706710394154 
 Bacharelado 
 Integral 
 9999.99 
 15 
 14.0 
 NaN 
 NaN 
 Campo Grande 
 MS 
 NTAwMjAwNDAyNzA0 
 Universidade Anhanguera - UNIDERP 
 CAMPO GRANDE - SEDE - Miguel Couto 
 706710 
 Medicina 
 740.22 
 726.46 
 NaN 
 NaN 
 
 
 104191210567043 
 Bacharelado 
 Noturno 
 9836.40 
 1 
 NaN 
 NaN 
 NaN 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Enfermagem 
 663.36 
 NaN 
 NaN 
 NaN 
 
 
 1002328574024 
 Bacharelado 
 Integral 
 9715.61 
 2 
 5.0 
 6.0 
 10.0 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 104191212798093 
 Bacharelado 
 Noturno 
 9689.34 
 3 
 2.0 
 NaN 
 NaN 
 Crateus 
 CE 
 MjMwNDAxODA0MTAz 
 Faculdade Princesa do Oeste - FPO 
 UNIDADE SEDE - São Vicente 
 1041912 
 Psicologia 
 651.00 
 652.22 
 NaN 
 NaN 
 
 
 65899611932754 
 Bacharelado 
 Integral 
 9674.34 
 4 
 1.0 
 5.0 
 2.0 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26

In [0]:
# Quantos cursos de medicina estão no dataset?
df[df['nome_curso'] == 'Medicina']['nome_curso'].count()

Out[15]: 124

In [0]:
# Em quais turnos temos cursos de medicina?
df['turno'].unique()

Out[16]: 0             Matutino
1    Curso a Distância
2           Vespertino
3             Integral
4              Noturno
Name: turno, dtype: object

In [0]:
df[(df['turno'] == 'Integral') & (df['nome_curso'] == 'Medicina')]['nome_curso'].count()

Out[17]: 123

In [0]:
df[(df['turno'] == 'Integral') & (df['nome_curso'] == 'Medicina') | (df['turno'] == 'Matutino') & (df['nome_curso'] == 'Medicina') ]['nome_curso'].count()

Out[18]: 124

In [0]:
# Mensalidades de cursos de medicina no Brasil
df_mensalidade.head()

reajuste 
 
 
 
 
 0 
 100 
 
 
 1 
 50 
 
 
 2 
 100 
 
 
 3 
 50 
 
 
 4 
 100

In [0]:
df_mensalidade.shape

Out[20]: (41447, 1)

In [0]:
ps.set_option('compute.ops_on_diff_frames', True)

In [0]:
df.reset_index(inplace = True)

In [0]:
df['mensalidade'] = df['mensalidade'] + df_mensalidade['reajuste']

In [0]:
# Removendo valores nulos
df.dropna(inplace = True)
df.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 1950 
 65982912800701 
 Bacharelado 
 Matutino 
 1561.29 
 1 
 1 
 1 
 30 
 Caucaia 
 CE 
 MjMwMzAxNjAzNzA5 
 Faculdade Terra Nordeste - FATENE 
 UNIDADE SEDE - Soledade 
 659829 
 Engenharia Civil 
 688.70 
 693.02 
 450.00 
 450.00 
 
 
 8440 
 65928650009893 
 Bacharelado 
 Noturno 
 980.00 
 1 
 1 
 3 
 2 
 Sao Gotardo 
 MG 
 MzEwNTAyMDYyMTA0 
 Faculdade de Ciências Gerenciais de São Gotard... 
 UNIDADE SEDE - Boa Esperança 
 659286 
 Direito 
 638.94 
 612.92 
 608.26 
 579.50 
 
 
 9458 
 658024423323 
 Bacharelado 
 Noturno 
 927.00 
 2 
 1 
 3 
 2 
 Dourados 
 MS 
 NTAwNDAxMDAzNzAy 
 Centro Universitário da Grande Dourados - Unig... 
 DOURADOS - Jardim 
 658024 
 Psicologia 
 631.32 
 609.34 
 618.58 
 557.68 
 
 
 11190 
 1072630671593 
 Bacharelado 
 Noturno 
 829.00 
 6 
 5 
 10 
 9 
 Juiz de Fora 
 MG 
 MzExMjA2NTM2NzAy 
 Faculdade Doctum de Juiz de Fora - DOCTUM 
 CAMPUS APOGEU - Centro 
 1072630 
 Direito 
 669.40 
 632.94 
 615.56 
 584.58 
 
 
 17979 
 69558711774842 
 Tecnológico 
 Vespertino 
 543.00 
 1 
 1 
 1 
 1 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 Faculdade da Amazônia Ocidental - FAAO 
 CAMPUS - RIO BRANCO - JARDIM PRIMAVERA - Jardi... 
 695587 
 Análise e Desenvolvimento de Sistemas 
 615.54 
 574.42 
 582.66 
 623.22

In [0]:
df.describe()

curso_id 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 campus_id 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 count 
 2.203000e+03 
 2203.000000 
 2203.000000 
 2203.000000 
 2203.000000 
 2203.000000 
 2.203000e+03 
 2203.000000 
 2203.000000 
 2203.000000 
 2203.000000 
 
 
 mean 
 4.376614e+13 
 881.605556 
 1.891058 
 2.082161 
 3.387199 
 7.190649 
 7.510660e+05 
 599.380390 
 574.097921 
 547.297204 
 521.452964 
 
 
 std 
 4.862265e+13 
 737.440291 
 1.969321 
 2.105756 
 4.020312 
 13.897084 
 4.202414e+05 
 53.194117 
 59.868058 
 58.971101 
 56.371352 
 
 
 min 
 1.649611e+08 
 194.000000 
 1.000000 
 1.000000 
 1.000000 
 1.000000 
 4.790000e+02 
 450.000000 
 450.000000 
 450.000000 
 450.000000 
 
 
 25% 
 6.589496e+11 
 500.000000 
 1.000000 
 1.000000 
 1.000000 
 2.000000 
 6.582050e+05 
 569.520000 
 542.220000 
 507.160000 
 450.000000 
 
 
 50% 
 1.002328e+13 
 773.830000 
 1.000000 
 1.000000 
 2.000000 
 3.000000 
 6.596950e+05 
 602.040000 
 583.320000 
 555.240000 
 526.540000 
 
 
 75% 
 1.040060e+14 
 1049.000000 
 2.000000 
 2.000000 
 4.000000 
 6.000000 
 1.054619e+06 
 634.600000 
 614.340000 
 588.160000 
 563.620000 
 
 
 max 
 2.008254e+14 
 9815.610000 
 34.000000 
 50.000000 
 74.000000 
 179.000000 
 2.008254e+06 
 781.200000 
 792.240000 
 776.360000 
 768.340000

In [0]:
#medicina
df_descricao = df[df['nome_curso'] == 'Medicina'].describe()
df_descricao

curso_id 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 campus_id 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 count 
 1.500000e+01 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 1.500000e+01 
 15.000000 
 15.000000 
 15.000000 
 15.000000 
 
 
 mean 
 4.407875e+13 
 8007.354667 
 2.800000 
 2.266667 
 4.466667 
 3.866667 
 7.599417e+05 
 753.249333 
 738.438667 
 746.453333 
 724.676000 
 
 
 std 
 4.349636e+13 
 1096.173682 
 1.859339 
 1.533747 
 2.587516 
 2.669047 
 2.510458e+05 
 13.109076 
 16.966989 
 14.295313 
 18.302904 
 
 
 min 
 1.373807e+11 
 6113.000000 
 1.000000 
 1.000000 
 1.000000 
 1.000000 
 1.373800e+05 
 735.840000 
 720.200000 
 729.720000 
 690.180000 
 
 
 25% 
 6.584619e+11 
 7130.000000 
 1.000000 
 1.000000 
 1.000000 
 2.000000 
 6.583450e+05 
 741.360000 
 729.020000 
 735.760000 
 715.100000 
 
 
 50% 
 6.579365e+13 
 8295.000000 
 2.000000 
 1.000000 
 5.000000 
 3.000000 
 6.600790e+05 
 754.980000 
 737.880000 
 743.140000 
 721.080000 
 
 
 75% 
 7.070341e+13 
 8608.540000 
 4.000000 
 3.000000 
 6.000000 
 5.000000 
 1.041237e+06 
 762.740000 
 741.100000 
 750.440000 
 732.460000 
 
 
 max 
 1.058879e+14 
 9815.610000 
 8.000000 
 5.000000 
 9.000000 
 10.000000 
 1.079920e+06 
 781.200000 
 792.240000 
 776.360000 
 768.340000

In [0]:
df_descricao_bolsa = df_descricao.drop(['mensalidade', 'curso_id','campus_id',"nota_integral_ampla", "nota_integral_cotas", "nota_parcial_ampla", "nota_parcial_cotas"], axis = 1)
df_descricao_bolsa = df_descricao_bolsa.drop(['count'], axis = 0)

A partir dessa tabela, podemos inferir algumas informações sobre as bolsas de estudo oferecidas nos cursos de graduação do conjunto de dados. A média de bolsas integrais concedidas por curso por meio de cotas é de 2.80, enquanto a média de bolsas integrais concedidas por ampla concorrência é de 2.27. Já a média de bolsas parciais concedidas por cotas é de 4.47, enquanto a média de bolsas parciais concedidas por ampla concorrência é de 3.87.

Além disso, podemos observar a variação desses números, representados pelo desvio padrão, e verificar que há uma grande variabilidade nas concessões de bolsas pelos cursos. Podemos também observar os valores mínimo e máximo e verificar que o número de bolsas concedidas pode variar de 1 a 8 para bolsas integrais por cotas, de 1 a 5 para bolsas integrais por ampla concorrência, de 1 a 9 para bolsas parciais por cotas e de 1 a 10 para bolsas parciais por ampla concorrência.

In [0]:
# Será que na Pandas dentro do PySpark também tem a opção de estilizar a informação?
df_descricao_bolsa.style.format('{:,.2f}').background_gradient(axis = 1)

bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 
 
 
 
 mean 
 2.80 
 2.27 
 4.47 
 3.87 
 
 
 std 
 1.86 
 1.53 
 2.59 
 2.67 
 
 
 min 
 1.00 
 1.00 
 1.00 
 1.00 
 
 
 25% 
 1.00 
 1.00 
 1.00 
 2.00 
 
 
 50% 
 2.00 
 1.00 
 5.00 
 3.00 
 
 
 75% 
 4.00 
 3.00 
 6.00 
 5.00 
 
 
 max 
 8.00 
 5.00 
 9.00 
 10.00

### Qual é a média da mensalidade dos cursos de medicina por estado?

In [0]:
# Média de mensalidades
df_medicina = df.query("nome_curso == 'Medicina'")
df_medicina.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28 
 
 
 71 
 65793650009334 
 Bacharelado 
 Integral 
 7316.79 
 2 
 3 
 4 
 8 
 Franca 
 SP 
 MzUwMjAxMjE2MjAw 
 Universidade de Franca - UNIFRAN 
 POLO NA SEDE - Parque Universitário 
 657936 
 Medicina 
 744.88 
 730.88 
 741.36 
 721.08 
 
 
 87 
 658461856524 
 Bacharelado 
 Integral 
 7009.97 
 4 
 3 
 9 
 5 
 Porto Velho 
 RO 
 MTEwMTAwMTAwMjA1 
 Centro Universitário São Lucas - UniSL 
 CAMPUS - PORTO VELHO - AREAL - Areal 
 658461 
 Medicina 
 742.72 
 720.20 
 729.72 
 711.68 
 
 
 107 
 66007913994844 
 Bacharelado 
 Integral 
 6498.55 
 1 
 1 
 1 
 4 
 Guarapuava 
 PR 
 NDEwODAyOTA5NDAx 
 Faculdade Campo Real - CAMPO REAL 
 BLOCO 1 - Santa Cruz 
 660079 
 Medicina 
 781.20 
 731.04 
 745.06 
 741.00 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82

In [0]:
df_medicina['mensalidade'].mean()

Out[30]: 8007.354666666666

In [0]:
# Mensalidade por cidade
df_medicina.query("turno == 'Integral' AND  cidade_busca == 'Sao Paulo'")

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 2 
 1002328574024 
 Bacharelado 
 Integral 
 9815.61 
 2 
 5 
 6 
 10 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28

In [0]:
# Mensalidade por estado
agrupado = df_medicina.query("turno == 'Integral' OR turno == 'Matutino'").groupby('uf_busca')['mensalidade'].mean()

In [0]:
df_agrupado = ps.DataFrame(agrupado)

In [0]:
df_agrupado

mensalidade 
 
 
 uf_busca 
 
 
 
 
 
 RO 
 7009.970000 
 
 
 SP 
 8477.466667 
 
 
 ES 
 6113.000000 
 
 
 PB 
 7512.700000 
 
 
 MG 
 7712.500000 
 
 
 PA 
 8608.540000 
 
 
 RJ 
 9123.070000 
 
 
 MA 
 8381.375000 
 
 
 AC 
 9774.340000 
 
 
 PR 
 7174.275000

In [0]:
# Ordem das informações
df_agrupado.sort_values(by = 'mensalidade', inplace = True, ascending = False)

In [0]:
df_agrupado

mensalidade 
 
 
 uf_busca 
 
 
 
 
 
 AC 
 9774.340000 
 
 
 RJ 
 9123.070000 
 
 
 PA 
 8608.540000 
 
 
 SP 
 8477.466667 
 
 
 MA 
 8381.375000 
 
 
 MG 
 7712.500000 
 
 
 PB 
 7512.700000 
 
 
 PR 
 7174.275000 
 
 
 RO 
 7009.970000 
 
 
 ES 
 6113.000000

In [0]:
# Gráfico de barras
df_agrupado = df_agrupado.reset_index()

Podemos ver que o Estado com maior média de mensalidades é o Acre e o menor é o Espírito Santo.

In [0]:
df_agrupado.plot.bar(x = 'uf_busca', y = 'mensalidade')

### Quais estados possuem um número maior de cotas?

In [0]:
# realizar uma query com SQL 
ps.sql('''
SELECT bolsa_integral_cotas, uf_busca
FROM {DF}
WHERE nome_curso = 'Medicina'
LIMIT 5;
''', DF = df)

bolsa_integral_cotas 
 uf_busca 
 
 
 
 
 0 
 2 
 SP 
 
 
 1 
 4 
 AC 
 
 
 2 
 4 
 RJ 
 
 
 3 
 2 
 PA 
 
 
 4 
 3 
 MA

In [0]:
# Cotas ordenadas por estado
cotas_ordenadas = ps.sql('''
SELECT uf_busca, SUM(bolsa_integral_cotas) AS Total_de_cotas
FROM {DF}
WHERE nome_curso = 'Medicina'
GROUP BY uf_busca
ORDER BY Total_de_cotas DESC
''', DF = df )

In [0]:
cotas_ordenadas.head()

uf_busca 
 Total_de_cotas 
 
 
 
 
 0 
 PB 
 8 
 
 
 1 
 MA 
 7 
 
 
 2 
 SP 
 6 
 
 
 3 
 RO 
 4 
 
 
 4 
 RJ 
 4

Em relação às cotas é o que possui mais cotas e o Espírito Santo é o sétimo.

In [0]:
# Visualizando as informações - Gráfico de barras
cotas_ordenadas.plot.bar(x = 'uf_busca', y = 'Total_de_cotas')

In [0]:
df_medicina['cidade_busca'].unique()

Out[43]: 0          Sao Luis
1       Joao Pessoa
2             Belem
3         Itaperuna
4           Vitoria
5            Franca
6         Sao Paulo
7        Guarapuava
8      Campo Mourao
9     Montes Claros
10       Rio Branco
11       Imperatriz
12       Vespasiano
13      Porto Velho
Name: cidade_busca, dtype: object

In [0]:
nomes_cidades = {
"Sao Luis": "Capital",
"Joao Pessoa": "Município",
"Belem": "Capital",
"Itaperuna": "Município",
"Vitoria": "Município",
"Franca": "Município",
"Sao Paulo": "Capital",
"Guarapuava": "Município",
"Campo Mourao": "Município",
"Montes Claros": "Município",
"Rio Branco": "Município",
"Imperatriz": "Município",
"Vespasiano": "Município",
"Porto Velho": "Município"
}

In [0]:
capitais = df_medicina['cidade_busca'].map(nomes_cidades)
capitais

Out[45]: 22       Capital
71     Município
87     Município
107    Município
10     Município
120    Município
80     Município
2        Capital
4      Município
13       Capital
21       Capital
42     Município
20     Município
60     Município
24     Município
Name: cidade_busca, dtype: object

In [0]:
df_medicina.insert(loc=8,column='tipo_cidade',value=capitais)

In [0]:
df_medicina.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 tipo_cidade 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 22 
 105596711902804 
 Bacharelado 
 Integral 
 8300.00 
 2 
 3 
 4 
 7 
 Capital 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 Faculdade das Américas - FAM 
 RUA AUGUSTA - Consolação 
 1055967 
 Medicina 
 735.84 
 724.70 
 738.42 
 723.28 
 
 
 71 
 65793650009334 
 Bacharelado 
 Integral 
 7316.79 
 2 
 3 
 4 
 8 
 Município 
 Franca 
 SP 
 MzUwMjAxMjE2MjAw 
 Universidade de Franca - UNIFRAN 
 POLO NA SEDE - Parque Universitário 
 657936 
 Medicina 
 744.88 
 730.88 
 741.36 
 721.08 
 
 
 87 
 658461856524 
 Bacharelado 
 Integral 
 7009.97 
 4 
 3 
 9 
 5 
 Município 
 Porto Velho 
 RO 
 MTEwMTAwMTAwMjA1 
 Centro Universitário São Lucas - UniSL 
 CAMPUS - PORTO VELHO - AREAL - Areal 
 658461 
 Medicina 
 742.72 
 720.20 
 729.72 
 711.68 
 
 
 107 
 66007913994844 
 Bacharelado 
 Integral 
 6498.55 
 1 
 1 
 1 
 4 
 Município 
 Guarapuava 
 PR 
 NDEwODAyOTA5NDAx 
 Faculdade Campo Real - CAMPO REAL 
 BLOCO 1 - Santa Cruz 
 660079 
 Medicina 
 781.20 
 731.04 
 745.06 
 741.00 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Município 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82

Já para quem prefere estudar nas grandes capitais temos 10 cursos em capitais de estados.

In [0]:
cotas_ordenadas = ps.sql('''
SELECT tipo_cidade, SUM(bolsa_integral_cotas) AS Total_de_cotas
FROM {DF}
GROUP BY tipo_cidade
ORDER BY Total_de_cotas DESC
''', DF=df_medicina)
cotas_ordenadas.plot.bar(x='tipo_cidade',y='Total_de_cotas')


É fundamental entender como a mensalidade de um curso se relaciona com a quantidade de bolsas oferecidas, pois isso pode influenciar significativamente a acessibilidade e a viabilidade financeira da educação superior para muitos estudantes. Então vamos responder a pergunta.

Como a mensalidade do curso se relaciona com a quantidade de bolsas?

In [0]:
# Como a mensalidade do curso se relaciona com a quantidade de bolsas? 
aux = df_medicina[['mensalidade','bolsa_integral_cotas','bolsa_integral_ampla','bolsa_parcial_cotas','bolsa_parcial_ampla','nota_integral_ampla', 'nota_integral_cotas', 'nota_parcial_ampla','nota_parcial_cotas']]

In [0]:
aux.corr().style.background_gradient(cmap="Reds",vmin=-1,vmax=1)

/databricks/spark/python/pyspark/sql/pandas/conversion.py:122: UserWarning:

toPandas attempted Arrow optimization because 'spark.sql.execution.arrow.pyspark.enabled' is set to true; however, failed by the reason below:
  Unable to convert the field pearson(__correlation_output__). If this column is not necessary, you may consider dropping it or converting to primitive type before the conversion.
Direct cause: Unsupported type in conversion to Arrow: MatrixUDT()
Attempting non-optimization as 'spark.sql.execution.arrow.pyspark.fallback.enabled' is set to true.



mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 mensalidade 
 1.000000 
 0.080124 
 0.085915 
 0.462838 
 0.281507 
 -0.438041 
 -0.042471 
 -0.490010 
 -0.556348 
 
 
 bolsa_integral_cotas 
 0.080124 
 1.000000 
 0.520982 
 0.451340 
 -0.120903 
 -0.155176 
 -0.360329 
 -0.247502 
 -0.064000 
 
 
 bolsa_integral_ampla 
 0.085915 
 0.520982 
 1.000000 
 0.182384 
 0.620009 
 -0.489539 
 -0.293681 
 -0.218251 
 0.084619 
 
 
 bolsa_parcial_cotas 
 0.462838 
 0.451340 
 0.182384 
 1.000000 
 0.195821 
 -0.468888 
 -0.253827 
 -0.546129 
 -0.469743 
 
 
 bolsa_parcial_ampla 
 0.281507 
 -0.120903 
 0.620009 
 0.195821 
 1.000000 
 -0.549894 
 -0.270162 
 -0.534464 
 -0.266389 
 
 
 nota_integral_ampla 
 -0.438041 
 -0.155176 
 -0.489539 
 -0.468888 
 -0.549894 
 1.000000 
 0.420823 
 0.409216 
 0.563932 
 
 
 nota_integral_cotas 
 -0.042471 
 -0.360329 
 -0.293681 
 -0.253827 
 -0.270162 
 0.420823 
 1.000000 
 0.402048 
 0.237963 
 
 
 nota_parcial_ampla 
 -0.490010 
 -0.247502 
 -0.218251 
 -0.546129 
 -0.534464 
 0.409216 
 0.402048 
 1.000000 
 0.702974 
 
 
 nota_parcial_cotas 
 -0.556348 
 -0.064000 
 0.084619 
 -0.469743 
 -0.266389 
 0.563932 
 0.237963 
 0.702974 
 1.000000

Não parece haver uma relação substancial em relação entre a quantidade de bolsas.

In [0]:
df.plot.scatter(x='mensalidade',y='bolsa_parcial_cotas', trendline="ols", trendline_color_override = 'red')

Existem outras relações que podem ser obversadas como as notas de quem tem cotas e de quem não possui.

In [0]:
df.plot.scatter(x='nota_parcial_ampla',y='nota_parcial_cotas', trendline="ols", trendline_color_override = 'red')

Como o período do curso afeta a mensalidade?

In [0]:
df[df['turno']=='Integral']['mensalidade'].plot.box()

In [0]:
df[df['turno']=='Matutino']['mensalidade'].plot.box()

Comparando os cursos de enfermagem com os de medicina vemos que os cursos de medicina são muito mais caros mesmo sendo de uma área correlata.

In [0]:
df[df['nome_curso']=='Medicina']['mensalidade'].plot.box()

In [0]:
df[df['nome_curso']=='Enfermagem']['mensalidade'].plot.box(fillcolor='red')

In [0]:
df_medicina['universidade_nome'].unique()

Out[57]: 0             Faculdades Integradas Pitágoras - FIP-MOC
1          Faculdade da Saúde e Ecologia Humana - FASEH
2                          Faculdade das Américas - FAM
3           Faculdade Metropolitana da Amazônia - FAMAZ
4     Faculdade de Ciências Médicas da Paraíba - FCM-PB
5                      Faculdade Redentor - FACREDENTOR
6                   Faculdade Barão do Rio Branco - FAB
7             Universidade Cidade de São Paulo - UNICID
8        Centro Universitário Integrado de Campo Mourão
9                     Faculdade Campo Real - CAMPO REAL
10               Centro Universitário São Lucas - UniSL
11              Faculdade Brasileira - MULTIVIX VITÓRIA
12          Universidade do Ceuma - Uniceuma - UNICEUMA
13                     Universidade de Franca - UNIFRAN
Name: universidade_nome, dtype: object

In [0]:
selecao = df_medicina[df_medicina['universidade_nome'].str.contains('Universidade')][['universidade_nome']]
selecao

universidade_nome 
 
 
 
 
 2 
 Universidade Cidade de São Paulo - UNICID 
 
 
 20 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 
 
 21 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 
 
 71 
 Universidade de Franca - UNIFRAN

In [0]:
selecao['universidade_nome'].unique().count()

Out[59]: 3

In [0]:
df_medicina['universidade_nome'].loc[42] = 'Centro Universitário Integrado de Campo Mourão - CUICM'
df_medicina['universidade_nome'].loc[42] 

Out[60]: 'Centro Universitário Integrado de Campo Mourão - CUICM'

In [0]:
separado = df_medicina['universidade_nome'].str.split(' - ',n=1,expand=True)
separado

0 
 1 
 
 
 
 
 22 
 Faculdade das Américas 
 FAM 
 
 
 71 
 Universidade de Franca 
 UNIFRAN 
 
 
 87 
 Centro Universitário São Lucas 
 UniSL 
 
 
 107 
 Faculdade Campo Real 
 CAMPO REAL 
 
 
 10 
 Faculdade Redentor 
 FACREDENTOR 
 
 
 120 
 Faculdade Brasileira 
 MULTIVIX VITÓRIA 
 
 
 80 
 Faculdades Integradas Pitágoras 
 FIP-MOC 
 
 
 2 
 Universidade Cidade de São Paulo 
 UNICID 
 
 
 4 
 Faculdade Barão do Rio Branco 
 FAB 
 
 
 13 
 Faculdade Metropolitana da Amazônia 
 FAMAZ 
 
 
 21 
 Universidade do Ceuma 
 Uniceuma - UNICEUMA 
 
 
 42 
 Centro Universitário Integrado de Campo Mourão 
 CUICM 
 
 
 20 
 Universidade do Ceuma 
 Uniceuma - UNICEUMA 
 
 
 60 
 Faculdade de Ciências Médicas da Paraíba 
 FCM-PB 
 
 
 24 
 Faculdade da Saúde e Ecologia Humana 
 FASEH

In [0]:
df_medicina.insert(loc=12,column='sigla',value=separado[1])

In [0]:
df_medicina.head()

curso_id 
 grau 
 turno 
 mensalidade 
 bolsa_integral_cotas 
 bolsa_integral_ampla 
 bolsa_parcial_cotas 
 bolsa_parcial_ampla 
 tipo_cidade 
 cidade_busca 
 uf_busca 
 cidade_filtro 
 sigla 
 universidade_nome 
 campus_nome 
 campus_id 
 nome_curso 
 nota_integral_ampla 
 nota_integral_cotas 
 nota_parcial_ampla 
 nota_parcial_cotas 
 
 
 
 
 2 
 1002328574024 
 Bacharelado 
 Integral 
 9815.61 
 2 
 5 
 6 
 10 
 Capital 
 Sao Paulo 
 SP 
 MzUxNTA2MTUwMzA4 
 UNICID 
 Universidade Cidade de São Paulo - UNICID 
 UNIVERSIDADE CIDADE DE SÃO PAULO - UNICID - SE... 
 1002328 
 Medicina 
 739.62 
 738.08 
 738.96 
 718.64 
 
 
 4 
 65899611932754 
 Bacharelado 
 Integral 
 9774.34 
 4 
 1 
 5 
 2 
 Município 
 Rio Branco 
 AC 
 MTIwMjAwNDAwNDAx 
 FAB 
 Faculdade Barão do Rio Branco - FAB 
 CAMPUS - RIO BRANCO - JARDIM EUROPA II - Jard... 
 658996 
 Medicina 
 758.32 
 723.94 
 734.92 
 711.26 
 
 
 10 
 65921511844324 
 Bacharelado 
 Integral 
 9123.07 
 4 
 4 
 5 
 5 
 Município 
 Itaperuna 
 RJ 
 MzMwMTAwMTAyMjA1 
 FACREDENTOR 
 Faculdade Redentor - FACREDENTOR 
 CAMPUS - ITAPERUNA - PRESIDENTE COSTA E SILVA... 
 659215 
 Medicina 
 741.06 
 729.02 
 730.98 
 722.82 
 
 
 13 
 104123711716094 
 Bacharelado 
 Integral 
 8608.54 
 2 
 1 
 7 
 2 
 Capital 
 Belem 
 PA 
 MTUwMzAwNzAxNDAy 
 FAMAZ 
 Faculdade Metropolitana da Amazônia - FAMAZ 
 FACULDADE METROPOLITANA DA AMAZÔNIA - FAMAZ - ... 
 1041237 
 Medicina 
 741.36 
 739.90 
 760.80 
 715.10 
 
 
 20 
 105887911921884 
 Bacharelado 
 Integral 
 8381.38 
 3 
 1 
 6 
 2 
 Município 
 Imperatriz 
 MA 
 MjEwMjAwOTA1MzAy 
 Uniceuma - UNICEUMA 
 Universidade do Ceuma - Uniceuma - UNICEUMA 
 UNIVERSIDADE DO CEUMA - UNICEUMA - CAMPUS IMPE... 
 1058879 
 Medicina 
 756.46 
 737.88 
 747.14 
 717.60

Em relação as faculdades, a que tem a melhor nota é a campo real.

In [0]:
# Nota média das faculdades
# Qual a nota média de cada faculdade?
notas_ordenadas = ps.sql('''
SELECT sigla, MEAN(nota_integral_ampla) AS nota_media
FROM {DF}
GROUP BY sigla
ORDER BY nota_media DESC
''', DF=df_medicina)
notas_ordenadas.plot.bar(x='sigla',y='nota_media')

Como a quantidade de bolsas variou nos últimos anos?

Fonte: https://pt.wikipedia.org/wiki/Programa_Universidade_para_Todos Acessado em: 16/03/2023

In [0]:
import pandas as pd

data = {
    'Ano': [2021,2020],
    'Bolsas_Integrais': [146.3, 182.9],
    'Bolsas_Parciais': [150.0, 237.4],
    'Bolsas_Total': [296.3, 420.3]
}

In [0]:
df_pandas = pd.DataFrame(data)

In [0]:
df_bolsas = ps.from_pandas(df_pandas)
df_bolsas

Ano 
 Bolsas_Integrais 
 Bolsas_Parciais 
 Bolsas_Total 
 
 
 
 
 0 
 2021 
 146.3 
 150.0 
 296.3 
 
 
 1 
 2020 
 182.9 
 237.4 
 420.3

Tivemos uma diminuição de bolsas em 2021

In [0]:
#df_bolsas.plot.bar(x='Ano',y=['Bolsas_Integrais','Bolsas_Parciais'],barmode='group')
df_bolsas.plot.bar(x='Ano', y=['Bolsas_Integrais', 'Bolsas_Parciais'], barmode='group', title='Tivemos uma diminuição de bolsas em 2021')


In [0]:
# Como puxar informações de um site?
url = 'https://pt.wikipedia.org/wiki/Ensino_superior_no_Brasil'

In [0]:
!pip install lxml

You should consider upgrading via the '/local_disk0/.ephemeral_nfs/envs/pythonEnv-ecccccf8-4777-45ec-a6c7-17ab9f15fc30/bin/python -m pip install --upgrade pip' command.


In [0]:
lista = ps.read_html(url)

In [0]:
len(lista)

Out[72]: 5

In [0]:
df_publicas = lista[0]
df_publicas

Classificação Nacional 
 Classificação Mundial 
 Universidade 
 Logo 
 sigla 
 tipo 
 web 
 Localização 
 
 
 
 
 0 
 1 
 72 
 Universidade de São Paulo 
 NaN 
 USP 
 Pública Estadual 
 www5.usp.br/ 
 São Paulo 
 
 
 1 
 2 
 236 
 Universidade Estadual de Campinas 
 NaN 
 UNICAMP 
 Pública Estadual 
 www.unicamp.br/ 
 São Paulo 
 
 
 2 
 3 
 278 
 Universidade Federal do Rio de Janeiro 
 NaN 
 UFRJ 
 Pública federal 
 ufrj.br/ 
 Rio de Janeiro 
 
 
 3 
 4 
 298 
 Universidade Federal de Minas Gerais 
 NaN 
 UFMG 
 Pública federal 
 ucb.edu.bo 
 Minas Gerais 
 
 
 4 
 5 
 345 
 Universidade Estadual Paulista 
 NaN 
 UNESP 
 Pública Estadual 
 www.unesp.br 
 São Paulo 
 
 
 5 
 6 
 427 
 Universidade Federal de Santa Catarina 
 NaN 
 UFSC 
 Pública federal 
 ufsc.br/ 
 Santa Catarina 
 
 
 6 
 7 
 592 
 Universidade de Brasília 
 NaN 
 UNB 
 Pública Estadual 
 www.unb.br/ 
 Distrito Federal 
 
 
 7 
 8 
 593 
 Universidade Federal do Paraná 
 NaN 
 UFPR 
 Pública federal 
 www.ufpr.br/ 
 Paraná 
 
 
 8 
 9 
 596 
 Universidade Federal Fluminense 
 NaN 
 UFF 
 Pública federal 
 www.uff.br/ 
 Rio de Janeiro 
 
 
 9 
 10 
 667 
 Universidade Federal de São Carlos 
 NaN 
 UFSCAR 
 Pública federal 
 www.ufscar.br/ 
 São Paulo

In [0]:
type(df_publicas)

Out[74]: pyspark.pandas.frame.DataFrame

In [0]:
df_publicas.sort_values(by=['Classificação Nacional'],inplace=True, ascending=False)

Caso a pessoa decida tentar uma universidade pública a USP é a que possui melhor nota, seguida pela Unicamp.

In [0]:
fig=df_publicas.plot.bar(x='sigla',y='Classificação Nacional',color=range(10))
fig.layout.coloraxis.showscale = False

fig.show()

In [0]:
### Salvar Informações
dfs_bolsas = df_bolsas.to_spark()

In [0]:
dfs_bolsas.show()

+----+----------------+---------------+------------+
| Ano|Bolsas_Integrais|Bolsas_Parciais|Bolsas_Total|
+----+----------------+---------------+------------+
|2021|           146.3|          150.0|       296.3|
|2020|           182.9|          237.4|       420.3|
+----+----------------+---------------+------------+



In [0]:
type(dfs_bolsas)

Out[79]: pyspark.sql.dataframe.DataFrame

In [0]:
local_do_arquivo = '/FileStore/tables/cursos_prouni.csv'

In [0]:
df_publicas.to_parquet('/FileStore/tables/publicas.parquet',partition_cols='Classificação Nacional')

In [0]:
df_publicas2 = ps.read_parquet('/FileStore/tables/publicas.parquet')
df_publicas2

Classificação Mundial 
 Universidade 
 Logo 
 sigla 
 tipo 
 web 
 Localização 
 Classificação Nacional 
 
 
 
 
 0 
 236 
 Universidade Estadual de Campinas 
 NaN 
 UNICAMP 
 Pública Estadual 
 www.unicamp.br/ 
 São Paulo 
 2 
 
 
 1 
 667 
 Universidade Federal de São Carlos 
 NaN 
 UFSCAR 
 Pública federal 
 www.ufscar.br/ 
 São Paulo 
 10 
 
 
 2 
 427 
 Universidade Federal de Santa Catarina 
 NaN 
 UFSC 
 Pública federal 
 ufsc.br/ 
 Santa Catarina 
 6 
 
 
 3 
 278 
 Universidade Federal do Rio de Janeiro 
 NaN 
 UFRJ 
 Pública federal 
 ufrj.br/ 
 Rio de Janeiro 
 3 
 
 
 4 
 298 
 Universidade Federal de Minas Gerais 
 NaN 
 UFMG 
 Pública federal 
 ucb.edu.bo 
 Minas Gerais 
 4 
 
 
 5 
 596 
 Universidade Federal Fluminense 
 NaN 
 UFF 
 Pública federal 
 www.uff.br/ 
 Rio de Janeiro 
 9 
 
 
 6 
 345 
 Universidade Estadual Paulista 
 NaN 
 UNESP 
 Pública Estadual 
 www.unesp.br 
 São Paulo 
 5 
 
 
 7 
 592 
 Universidade de Brasília 
 NaN 
 UNB 
 Pública Estadual 
 www.unb.br/ 
 Distrito Federal 
 7 
 
 
 8 
 593 
 Universidade Federal do Paraná 
 NaN 
 UFPR 
 Pública federal 
 www.ufpr.br/ 
 Paraná 
 8 
 
 
 9 
 72 
 Universidade de São Paulo 
 NaN 
 USP 
 Pública Estadual 
 www5.usp.br/ 
 São Paulo 
 1